### PR TITLE
Bugzilla/558032/title block to merge

### PR DESCRIPTION
**Pull request by g-eco**

### CODE REVIEW

File: core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java [hunk 1]
```diff
@@ -270,7 +283,23 @@ public static DAnnotation addTitleBlockLine(DDiagram diagram, DAnnotation titleB
...
+    titleBlock.getReferences().add(position + 1, line);
+    return line;
+  }
+  
+  /**
+   * Add new Title Block to diagram to the given position
+   * and create the columns in the line also
+   * 
+   * @param diagram
+   * @return the added Title Block
```
Comment (MalinaStefaniaStoicanescu) [inline core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java]: the added Title Block line

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java [hunk 2]
```diff
@@ -456,16 +457,24 @@ public void deleteTitleBlock(DAnnotation titleBlock, DDiagram diagram) {
   /**
    * this function removes a line of the Title Block
    * 
-   * @param titleBlockLine:
-   *          the annotation line of a title block
+   * @param cell:
+   *          removed the entire line that contains the annotation cell
```
Comment (MalinaStefaniaStoicanescu) [inline core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: this is an explanation of what the method does, not for the parameter (?)

Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: OK, will update.

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java [hunk 3]
```diff
@@ -456,16 +457,24 @@ public void deleteTitleBlock(DAnnotation titleBlock, DDiagram diagram) {
...
-    toRemove.add(titleBlockLine);
-    toRemove.addAll(TitleBlockHelper.getTitleBlockCells(titleBlockLine));
-    diagram.getEAnnotations().removeAll(toRemove);
+  public void removeLineOfTitleBlock(DAnnotation cell, DDiagram diagram) {
+    if (TitleBlockHelper.isTitleBlockCell(cell)) {
+      DAnnotation titleBlockContainer = TitleBlockHelper.getParentTitleBlock(cell, diagram);
+
+      // remove the line of the cell we clicked on
+      int indexLine = TitleBlockHelper.getLineIndexOfCell(cell, titleBlockContainer);
+      DAnnotation titleBlockCell = TitleBlockHelper.getTitleBlockLines(titleBlockContainer).get(indexLine);
```
Comment (MalinaStefaniaStoicanescu) [inline core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: rename in titleBlockLine

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java [hunk 4]
```diff
@@ -323,7 +346,18 @@ public void clearCellContent(EObject object) {
         if (element.getTarget() instanceof DAnnotation) {
           DAnnotation annotation = (DAnnotation) element.getTarget();
           if (annotation.getSource() != null && annotation.getSource().equals(type)) {
-            list.add(annotation);
+            if(type.equals(TitleBlockHelper.ELEMENT_TITLE_BLOCK)) {
+              DNodeContainer node = (DNodeContainer) element;
+              if(!node.getIncomingEdges().isEmpty()) {
+                DEdge edge = node.getIncomingEdges().iterator().next();
+                EdgeTarget sN = edge.getSourceNode();
```
Comment (MalinaStefaniaStoicanescu) [inline core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: rename variable

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java [hunk 5]
```diff
@@ -547,73 +673,122 @@ public boolean isWrapperOfPrimitiveType(EObject object) {
...
+              IMappingNameConstants.DT_TITLE_BLOCK_COLUMN_CONTAINER);
+          AbstractDNode nodeCol = DiagramServices.getDiagramServices().createDNodeListElement(mappingCol,
+              annotationCol, (DragAndDropTarget)nodeLine, diagram);
+          
+          for(EObject objCell : annotationCol.getReferences()) {
+            
+            // for each cell
+            NodeMapping mappingCell = DiagramServices.getDiagramServices().getNodeMapping(diagram,
+                IMappingNameConstants.DT_TITLE_BLOCK_CELL);
+            AbstractDNode nodea = DiagramServices.getDiagramServices().createDNodeListElement(mappingCell, objCell, (DragAndDropTarget) nodeCol, diagram);
```
Comment (MalinaStefaniaStoicanescu) [inline core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: rename variable

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java [hunk 6]
```diff
@@ -547,73 +673,122 @@ public boolean isWrapperOfPrimitiveType(EObject object) {
...
+        DiagramServices.getDiagramServices().createEdge(mapping, (EdgeTarget) elem, node, titleBlock);
       }
-      return DiagramServices.getDiagramServices().createDNodeListElement(mapping, titleBlock,
-          (DragAndDropTarget) context, diagram);
-    }
-    if (null != titleBlock.getDetails().get(TitleBlockHelper.VISIBILITY)) {
-      titleBlock.getDetails().put(TitleBlockHelper.VISIBILITY, TitleBlockHelper.TRUE);
     }
-    return DiagramServices.getDiagramServices().createNode(mapping, titleBlock, (DragAndDropTarget) context, diagram);
+    return node;
```
Comment (MalinaStefaniaStoicanescu) [inline core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: node is not really suggestive

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java [hunk 7]
```diff
@@ -197,7 +215,11 @@ public void createDiagramTitleBlock(DDiagram diagram) {
   public void createElementTitleBlock(DDiagramElement elementView, DDiagram diagram) {
     DAnnotation titleBlock = TitleBlockHelper.addElementTitleBlock(diagram, elementView);
     DAnnotation line = TitleBlockHelper.addTitleBlockLine(diagram, titleBlock);
-    TitleBlockHelper.addTitleBlockCell(diagram, line, DEFAULT_CELL_NAME, DEFAULT_CELL_CONTENT);
+    DAnnotation cell = TitleBlockHelper.addTitleBlockCell(diagram, line, DEFAULT_CELL_NAME, DEFAULT_CELL_CONTENT);
+    Object object = getCellContent(diagram, cell, diagram, titleBlock);
```
Comment (MalinaStefaniaStoicanescu) [inline core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: rename cellContent?

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java (hunk 3 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: ok.

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java (hunk 4 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: yes :)))

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java (hunk 6 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: accepted.

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java (hunk 5 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: accepted.

File: core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java (hunk 1 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java]: Yes, I will update the description.

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java (hunk 7 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: Accepted.

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java [hunk 8]
```diff
@@ -323,7 +346,18 @@ public void clearCellContent(EObject object) {
         if (element.getTarget() instanceof DAnnotation) {
           DAnnotation annotation = (DAnnotation) element.getTarget();
           if (annotation.getSource() != null && annotation.getSource().equals(type)) {
-            list.add(annotation);
+            if(type.equals(TitleBlockHelper.ELEMENT_TITLE_BLOCK)) {
+              DNodeContainer node = (DNodeContainer) element;
+              if(!node.getIncomingEdges().isEmpty()) {
+                DEdge edge = node.getIncomingEdges().iterator().next();
```
Comment (MalinaStefaniaStoicanescu) [inline core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: could use .get(0) instead  of iterator:-??

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java [hunk 9]
```diff
@@ -494,41 +572,61 @@ public void removeColumnOfTitleBlock(DAnnotation cell, DDiagram diagram) {
...
+  }
+  
   /**
    * convert a primitive object to an EObject using SimpleAnyType
    * 
    * @param object
    * @return EObject
    */
-  private EObject convertToEObject(Object object) {
+  private EObject convertToEObjectUsingSimpleAnyTyoe(Object object) {
```
Comment (MalinaStefaniaStoicanescu) [inline core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java:629]: Tipe*

Comment (MalinaStefaniaStoicanescu) [inline core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: Type

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/TitleBlockRefreshExtensionProvider.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+import org.eclipse.sirius.diagram.DDiagram;
+import org.eclipse.sirius.diagram.business.api.refresh.IRefreshExtension;
+import org.eclipse.sirius.diagram.business.api.refresh.IRefreshExtensionProvider;
+
+public class TitleBlockRefreshExtensionProvider implements IRefreshExtensionProvider {
+
+  private static final TitleBlockRefreshExtension REFRESH_EXTENSION = new TitleBlockRefreshExtension();
+
+  public TitleBlockRefreshExtensionProvider() {
```
Comment (MalinaStefaniaStoicanescu) [inline core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/TitleBlockRefreshExtensionProvider.java]: remove constructor

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/TitleBlockRefreshExtensionProvider.java [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+  public IRefreshExtension getRefreshExtension(DDiagram viewPoint_p) {
+    return REFRESH_EXTENSION;
+  }
+
+  /**
+   * @see org.eclipse.sirius.business.api.refresh.IRefreshExtensionProvider#provides(org.eclipse.sirius.DDiagram)
+   */
+  public boolean provides(DDiagram viewPoint_p) {
+    return true;
+        //(IDiagramNameConstants.OPERATIONAL_ENTITY_BLANK_DIAGRAM_NAME.equals(viewPoint_p.getDescription().getName()));
```
Comment (MalinaStefaniaStoicanescu) [inline core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/TitleBlockRefreshExtensionProvider.java]: basic mistake..

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/TitleBlockRefreshExtensionProvider.java [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
```
Comment (MalinaStefaniaStoicanescu) [inline core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/TitleBlockRefreshExtensionProvider.java]: remove empty line

File: core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java (hunk 1 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java]: Done.

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java (hunk 7 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: Done. I gave up this line, no more needed to getCellContent.

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java (hunk 8 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: Done.

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java (hunk 4 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: Done

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java (hunk 2 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: Done

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java (hunk 3 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: Done.

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java (hunk 9 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: Done. Removed the method, not used!

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java (hunk 5 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: Done. Removed the local nodea.

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java (hunk 6 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java]: Done. Returns void now.

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/TitleBlockRefreshExtensionProvider.java (hunk 10 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/TitleBlockRefreshExtensionProvider.java]: Done

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/TitleBlockRefreshExtensionProvider.java (hunk 11 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/TitleBlockRefreshExtensionProvider.java]: removed comment.

**Review by MalinaStefaniaStoicanescu — APPROVED** [review]

**Pull request merged**